### DISCUSSION TITLE
Accessing the Tooltip provider anywhere

### DISCUSSION THREAD

**Discussion by Raid55** [radix-ui/primitives#1631] [Help] [answered]
Hello 👋 

I am trying to use radix's tooltip component in sync with another tooltip library that is forced on us. The problem arises from the other tooltip library popping up at the same time as the radix popup because of some overlap in UI.

I could prop drill in a function that when radix is open it would send up a state change to then trickle back down to the other tooltip and then turn it off while its on, but I dont like this approach. 

Since the both tooltips are under the Tooltip.Provider is there a way to plugin-to the provider and get this open/closed state from any child of the provider? I look at the source and saw `createTooltipContext`  with the open/close logic a bit further down. I did not look into it much so I figure i'd ask.

**Comment by andy-hook** [comment, marked as answer]
@Raid55 There isn't a way to hook into the existing provider. I'd suggest using your own context in combination with the [controlled props](https://www.radix-ui.com/docs/primitives/components/tooltip#root) to store the value in a similar fashion to what you describe:

> it would send up a state change to then trickle back down to the other tooltip and then turn it off while its on, but I dont like this approach.
